x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx

```diff
@@ -237,6 +240,16 @@ const CreateRulePageComponent: React.FC = () => {
     dataViewId: defineStepData.dataViewId,
   });
```

Comment:
We should explore unmounting the unused components, it would definitely help the render times. However I've had issues so far with allowing components to be unmounted because the form data gets cleared out (even though the form itself doesn't unmount).

Part of the problem is that `StepDefineRuleComponent` uses `useFormData` so the whole define step re-renders every time any field changes. I'll try eliminating `useFormData` from the define step similar to how I did for the about step and that may help a lot with the re-rendering on every key press in any field.

Reply:
I was able to remove the dependency on `useFormData` from `StepDefineRuleComponent` so it no longer re-renders the whole form on every keypress, e.g. when typing in the query bar. Changing values in fields that are passed into `StepDefineRuleComponent`, e.g. `index`, does still cause the whole form to re-render, but we can keep looking for ways to reduce the number of these renders.